src/Storage/Bucket.php

```diff
@@ -671,4 +739,28 @@ public function name()
     {
         return $this->identity['bucket'];
     }
+
+    /**
+     * Returns whether the bucket with the given file prefix is writable.
+     * Tries to create a temporary file as a resumable upload which will
+     * not be completed (and cleaned up by GCS).
+     *
+     * @param  string $file Optional file to try to write.
+     * @return boolean
+     */
+    public function isWritable($file = null)
```

Comment:
It's possible, but would add a lot of complexity and a maintenance nightmare. The ACL rules could be very complex and attempting to resolve it on the client-side is not worth it in my opinion. We would cross reference the connection's user against the bucket roles for WRITER or OWNER as well as against the default permissions.

If, for whatever reason, the storage team decides to change permission structure/logic, then a client-side implementation would be obsolete.

Instead, we can just try an upload without completing it, and seeing if we're allowed to or not.